pkg/types/validation/installconfig.go

```diff
@@ -596,13 +596,20 @@ func validateNetworkingClusterNetworkMTU(c *types.InstallConfig, fldPath *field.
 	return allErrs
 }
 
-func validateControlPlane(platform *types.Platform, pool *types.MachinePool, fldPath *field.Path) field.ErrorList {
+func validateControlPlane(platform *types.Platform, pool *types.MachinePool, isSingleNodeOpenShift bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if pool.Name != types.MachinePoolControlPlaneRoleName {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("name"), pool.Name, []string{types.MachinePoolControlPlaneRoleName}))
 	}
-	if pool.Replicas != nil && *pool.Replicas == 0 {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be positive"))
+	if pool.Replicas != nil {
+		switch {
+		case *pool.Replicas == 0:
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be positive"))
+		case *pool.Replicas == 1 && !supportedSingleNodePlatform(isSingleNodeOpenShift, platform):
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "Single Node OpenShift(SNO) not supported by this install method for this platform"))
+		case *pool.Replicas != 3:
```

Comment:
nit: these validation functions can get pretty unreadable. I would extracting this into a separate function, say `ValidateControlPlaneReplicaCount` and use the same pattern as `ValidateMachinePool` so this function looks like:

```go
allErrs = append(allErrs, ValidateControlPlaneReplicaCount(platform, pool, fldPath)...)
allErrs = append(allErrs, ValidateMachinePool(platform, pool, fldPath)...)
```